parachain/pallets/ethereum-beacon-client/src/benchmarking/mod.rs

```diff
@@ -4,35 +4,69 @@ use crate::Pallet as EthereumBeaconClient;
 use frame_benchmarking::{benchmarks, impl_benchmark_test_suite, whitelisted_caller};
 use frame_system::RawOrigin;
 
-mod data;
+#[cfg(feature = "minimal")]
+mod data_minimal;
+#[cfg(feature = "minimal")]
+use data_minimal::*;
+
+#[cfg(not(feature = "minimal"))]
+mod data_mainnet;
+#[cfg(not(feature = "minimal"))]
+use data_mainnet::*;
 
 benchmarks! {
 	sync_committee_period_update {
+
 		let caller: T::AccountId = whitelisted_caller();
 
-		let initial_sync_data = data::initial_sync();
-		EthereumBeaconClient::<T>::initial_sync(initial_sync_data.clone()).unwrap();
+		let initial_sync_data = initial_sync();
+
+		EthereumBeaconClient::<T>::initial_sync(initial_sync_data.clone())?;
 
-		let update = data::sync_committee_update();
+		let sync_committee_update = sync_committee_update();
 
-	}: sync_committee_period_update(RawOrigin::Signed(caller.clone()), update.clone())
+		//initialize SyncCommittees with period in sync_committee_update
+		LatestSyncCommitteePeriod::<T>::set(EthereumBeaconClient::<T>::compute_current_sync_period(
+				sync_committee_update.attested_header.slot,
+			));
+		SyncCommittees::<T>::insert(
+			EthereumBeaconClient::<T>::compute_current_sync_period(
+				sync_committee_update.attested_header.slot,
+			),
+			initial_sync_data.current_sync_committee,
+		);
+
+	}: sync_committee_period_update(RawOrigin::Signed(caller.clone()), sync_committee_update.clone())
 	verify {
-		assert!(<SyncCommittees<T>>::get(update.sync_committee_period+1).pubkeys.len() > 0);
+		assert!(<SyncCommittees<T>>::get(sync_committee_update.sync_committee_period+1).pubkeys.len() > 0);
 	}
 
 	import_finalized_header {
 		let caller: T::AccountId = whitelisted_caller();
 
-		EthereumBeaconClient::<T>::initial_sync(data::initial_sync()).unwrap();
+		let initial_sync_data = initial_sync();
+
+		EthereumBeaconClient::<T>::initial_sync(initial_sync_data.clone())?;
+
+		let finalized_header_update = finalized_header_update();
 
-		let sync_update: SyncCommitteePeriodUpdate<T::MaxSignatureSize, T::MaxProofBranchSize, T::MaxSyncCommitteeSize> = data::sync_committee_update();
-		SyncCommittees::<T>::set(sync_update.sync_committee_period+1, sync_update.next_sync_committee);
+		SyncCommittees::<T>::insert(
+			EthereumBeaconClient::<T>::compute_current_sync_period(
+				finalized_header_update.attested_header.slot,
+			),
+			initial_sync_data.current_sync_committee,
+		);
 
-		let finalized_header = data::finalized_header_update();
+		//initialize LatestFinalizedHeaderState with parent slot of finalized_header_update
```

Comment:
```suggestion
		//initialize LatestFinalizedHeaderState with slot of finalized_header_update
```

Parent doesn't make sense here since the beacon header here is before the finalized header.